streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -204,6 +204,8 @@ class StreamingContext private[streaming] (
 
   private var shutdownHookRef: AnyRef = _
 
+  conf.getOption("spark.streaming.checkpointDir").foreach(checkpoint)
```

Comment:
Can you rename this configuration to `spark.streaming.checkpoint.directory`. This is to maintain cleanliness in the future if we add more configuration related to checkpointing (`spark.streaming.checkpoint.*`)